test/parallel/test-buffer-isutf8.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+require('../common');
+const assert = require('assert');
+const { isUtf8, Buffer } = require('buffer');
+const { TextEncoder } = require('util');
+
+const encoder = new TextEncoder();
+
+assert.strictEqual(isUtf8(encoder.encode('hello')), true);
+assert.strictEqual(isUtf8(encoder.encode('ğ')), true);
+assert.strictEqual(isUtf8(Buffer.from([])), true);
```

Comment:
Why does a zero length buffer return true? I would expect this to be false.

Reply:
Because it does not include an invalid code point. Is there a similar Node function that have a different behavior than this?

Reply:
But the stated description of the API is, "This function is used to check if input contains UTF-8 code points"... An empty buffer does not contain UTF-8 code points so it really can't return true. Other methods we have that accept ArrayBuffer or TypedArray, with the exception of Web Streams which have specifically defined handling for detached, will treat those as indistinguishable from a zero-length input.

Reply:
Hm.. Thats correct. What do you recommend?

Reply:
I would just follow up with an additional pr that returned false for zero-length, removing the detached check and error entirely.

Reply:
Thanks! Unfortunately, I’m dealing with some health issues and will not be available to fix this for a couple of days. Appreciate if you (or someone) can fix it.

Reply:
In my opinion it should _not_ be changed. It should return `true`. It's like the empty string which is valid UTF-8.

Reply:
The challenge there is that with that logic the empty buffer would pass any encoding check. isASCII? Yes. isUTF16le? Yes. IsUTF32be? Yes. Is Shift-JIS? Yes.... Which just simply isn't useful. If you want the inverse check, isInvalidUtf8() then implement *that*.

Reply:
I created a pull request: https://github.com/nodejs/node/pull/45973

Reply:
The UTF-8 RFC specifies...

```
An octet sequence is valid UTF-8 only if it matches the
   following syntax, which is derived from the rules for encoding UTF-8
   and is expressed in the ABNF of [[RFC2234](https://www.rfc-editor.org/rfc/rfc2234)].

   UTF8-octets = *( UTF8-char )
...
```

Reference: https://www.rfc-editor.org/rfc/rfc3629

So UTF-8 explicitly,  by its ABNF, includes the empty string. 

Note that, in general, from a non-empty buffer alone, we cannot determine uniquely the character encoding. A BOM may help but UTF-8 is BOM-less.

A string of bytes may be interpreted under different encodings... and in some cases, it is by design. Thus, for example, ASCII buffers are always valid UTF-8 and Latin1 buffers (by design).

Reply:
> The challenge there is that with that logic the empty buffer would pass any encoding check.

Yes, I think it makes sense and that is how it works in some other popular programming languages.

- https://onecompiler.com/ruby/3ysxqsm7q
- https://onecompiler.com/go/3ysxqv4gv